src/daft-table/src/repr_html.rs

```diff
@@ -0,0 +1,139 @@
+use daft_core::datatypes::ExtensionArray;
+use daft_core::{prelude::DataType, series::Series};
+
+pub fn html_value(s: &Series, idx: usize) -> String {
```

Comment:
https://github.com/Eventual-Inc/Daft/pull/2806 moves some of the display functionality out, we should reconcile with that!

Reply:
One of the patterns that I did in #2806 is that I made a `StrValue` trait. What if we do the same for `HTMLValue` and somehow register the HTML function (Datatype -> Function) for ImageArray from the image crate. Let's say with the `inventory` [crate](https://github.com/dtolnay/inventory)?

Reply:
A benefit of that is that we can preserve html rendering for a series

Reply:
Initially, i was trying to make a `HTMLValue` trait but was running into some issues that I think were addressed in #2806. Will take another look!

Reply:
@samster25 I took another look, and I don't think this is possible with our current architecture. We would have to fully move `ImageArray` out of core and into `daft-image` which comes with it's own host of problems, and is a SIGNIFICANTLY larger refactor. 

the problem is that the `html_value` implementation relies on the `image` crate, which is the main thing we want to pull out of core. 

I'm not super familiar with the inventory crate, but I don't see how it is supposed to solve the problem. 

If the goal is to remove the `daft-image -> daft-table` dependency, we could simple move the `repr_html` logic into python, which needs all dependencies anyways.